### ISSUE TITLE
Questions about coming version

### ISSUE THREAD

**Issue by shlomiken** [villasv/turbine#132]
Hi there
i really like this project - its very comprehensive and time saver.
some questions
1. i can see master is still on version 10.2 - do you have estimate when develop branch will be merged to master. will it be 10.4 ?
2. is it possible to upgrade/update my template and deployment without deleting the stack and recreating it ?
3. i need all workers and probably also webserver to have AWS credentials , preferably through ~/.aws/credentials file , or environment variables - how can this be achieved ?.
4. why all the complicated network configuration ? creating VPC and other stuff , cant it be really simple with existing VPC 
thanks
Shlomi

**Comment by villasv** [comment]
Hi @shlomiken, thanks for the kind words.

1. Yes, and in fact, it might be merged with 1.10.5 because it's already on RC
2. Very unlikely because the template files were split, so CFN isn't able to preserve the RDS instance (among the many other difficulties)
3. You could manage those with CodeDeploy, thouch I'd strongly recommend that you don't handle those files directly and instead rely on IAM roles
4. The previous version was indeed much simpler in terms of networking, but was hard to adapt for demanding security practices in production. The new version has two separate stacks: `master` which also deploys a standardized vpc stack and uses it, `cluster` which allows you to provide your own vpc information.

**Comment by shlomiken** [comment]
thanks @villasv .
regarding 3 - if someone for example use boto to work with S3 that requires credentials (access key and secret) , you cannot just use an IAM role and say - yeh trust me its me.:-)
for boto to work AWS credentials needs to be presented in home folder or environment variables.
i don't see how copy them to /airflow folder will help (and of course i don't want my code to look for the files this should be external to the code)

**Comment by shlomiken** [comment]
@villasv - another question , i also don't see how the requirements are installed on worker.
i saw in one PR you mentioned that webserver and scheduler should be done manually but i don't see the code that does this for a worker .

**Comment by villasv** [comment]
> if someone for example use boto to work with S3 that requires credentials (access key and secret) , you cannot just use an IAM role and say - yeh trust me its me.:-)

Boto3 automatically searches for credentials assuming the role present on the instance metadata, as documented [here](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#configuring-credentials). Even if you have multiple credentials, you can still instruct boto to ignore those and use the role for the instance.

> i also don't see how the requirements are installed on worker

Are you referring to the current master branch, current develop branch or some other version?

**Comment by shlomiken** [comment]
ok i wasn't aware of this assume role - but reading now , its still looks like a file at ~/.aws/credentials and config should present ..and they even point to the profile which have the credentials, so i don't see the point of having the assume role here
from their docs
```
# In ~/.aws/credentials:
[development]
aws_access_key_id=foo
aws_access_key_id=bar

# In ~/.aws/config
[profile crossaccount]
role_arn=arn:aws:iam:...
source_profile=development
```
regarding requirements install - i meant the develop branch (i want to use the cluster template)

**Comment by villasv** [comment]
Ok, in the `develop` branch we don't install the `requirements.txt` at all, not on workers and not on webserver and not on scheduler.

You can add your own pip install command to the deployment script, like [here](https://github.com/villasv/aws-airflow-stack/blob/master/src/airflow/scripts/cdapp_start.sh), so that before starting the service you'll be sure to have the latest packages.

Package instalation was removed from the stack because apart from Airflow and other infrastructure requirements, project dependencies are in the scope of project deployment (and not stack creation).

**Comment by shlomiken** [comment]
@villasv - sorry for late response , was on holidays for nearly a month.
can you please help me to deploy this template (develop branch) 
i want to deploy it to existing VPC (so i guess cluster template) - and when i try to deploy the template from develop branch - i get a stack form which is empty for defaults , some of the fields i have no idea what to fill in for example AllowedWebBlock, PrivateSubnet1AID, PrivateSubnet2AID.
i remember when i tried to deploy master - it had some sensible defaults , now i have no idea what to put in.

**Comment by villasv** [comment]
Hi @shlomiken. Most of these networking parameters will not have defaults, but we definitely need some sensible descriptions for them.

Now that the develop branch was merged into master, I think we should close this issue and open more specific ones for problems you find as you go.

I'll go ahead and create one for adding descriptions to the `turbine-cluster.template` parameters.
